libdistopia/src/distopia.cpp

```diff
@@ -685,6 +685,169 @@ namespace distopia {
             }
         }
 
+        template <typename V>
+        HWY_INLINE void LoadInterleavedIdx(const unsigned int *idx, const float *src,
+                                           V &x_dst, V &y_dst, V &z_dst) {
+            hn::ScalableTag<float> d;
+            hn::ScalableTag<int> d_int;
+            auto vidx = hn::LoadU(d_int, (int*)idx);
+            auto Vthree = hn::Set(d_int, 3);
+            auto Vone = hn::Set(d_int, 1);
+            // convert indices to 3D coordinate indices, i.e. index 10 at pointer 30 etc
+            vidx = hn::Mul(vidx, Vthree);
+            x_dst = hn::GatherIndex(d, src, vidx);
```

Comment:
This isn't actually that good, I imagine for performance you instead would want to do loads of xyz vectors (with junk) then a transpose, but you'd have to write this for all vector widths, at which point maybe you'd be better off doing a contribution upstream (`GatherIndexInterleaved3D`...) rather than writing so much code in the bullet farm